### ISSUE TITLE
Avatar: Select visible area

### ISSUE THREAD

**Issue by ovalseven8** [qTox/qTox#2384]
Hello,

something I miss in qTox and would improve the usability a lot is that you're able to select the visible area.

What I mean:
Assume you have a picture that is too big for an avatar. But still you'd like to use it.
So did Bob => ![Bob's Avatar](https://cloud.githubusercontent.com/assets/8258609/10416693/048f39f2-7020-11e5-9032-21064576b629.GIF)

But in qTox it looks not how Bob likes it (![Bob's Avatar in qTox](https://cloud.githubusercontent.com/assets/8258609/10416696/367a8d90-7020-11e5-95c2-93ee5d15fa49.png)). So, what is needed is something like a popup that says:
'Your avatar is too big for qTox. Please select the visible area.'
Then, Bob is able to select the area he wants to see in the avatar. And Bob's head, for example, is no longer cut off.

**Comment by zetok** [comment]
While it would be useful, I'm not convinced that it should be a part of qTox.

How would it function UI-wise?

**Comment by ovalseven8** [comment]
I was thinking about something like that:
![Screenshot 1](https://cloud.githubusercontent.com/assets/8258609/10417210/16002b34-7033-11e5-8543-23de42c9adeb.png) -> ![Screenshot 2](https://cloud.githubusercontent.com/assets/8258609/10417211/192c748e-7033-11e5-9ff8-45efc24a3f22.png)

So that you can drag the visible region with the mouse.

**Comment by ghost** [comment]
I think that it improves the usability (Tox project goal), it'd be nice to have such feature included.

> not convinced that it should be a part of qTox

If the client manages the user avatar, then it should help making it look the way it should, doesn't it?

**Comment by ProMcTagonist** [comment]
![qtox profile image resizing mockup](https://cloud.githubusercontent.com/assets/4358079/10418688/df4d5c68-7016-11e5-9f00-76f263770b8b.png)

In this concept, clicking and dragging on Bob himself moves him around behind the viewport, and the profile image is whatever we see through it. (This requires qTox saving the full image internally so it can be changed later without reselecting.)

There's a change button instead of clicking the image to change, so it doesn't happen accidentally while dragging. 

Needs a little more rearranging. Somebody better at UI should oversee this.

**Comment by zetok** [comment]
@ProMcTagonist ooh, that's nice/awesome

**Comment by ghost** [comment]
:+1: @ProMcTagonist 

I wonder if the zoom slider and buttons could be hidden while the user isn't changing/editing the profile picture, to keep the interface cleaner. Something like

Normal state, just like now

![001](https://cloud.githubusercontent.com/assets/6382342/10425448/0fccd442-70d8-11e5-9b6e-5604e1985d54.png)

On hover, change the appearance (it can be not-ugly, I'm just bad at Gimp). Clicking anywhere but the edit button opens the file chooser.

![002](https://cloud.githubusercontent.com/assets/6382342/10425449/1459e158-70d8-11e5-8fca-11681cc0bfe7.png)

Change the edit button opacity to 100% on hover

![003](https://cloud.githubusercontent.com/assets/6382342/10425451/19a36f08-70d8-11e5-8f93-374080dece03.png)

Display the image editing mode after clicking the edit button or after setting a new profile picture

![004](https://cloud.githubusercontent.com/assets/6382342/10425505/688c581e-70d8-11e5-875b-4958960ba5af.png)

Plus tooltips for the image ("click to select a new profile picture") and the edit button ("click to edit your profile picture").

**Comment by ovalseven8** [comment]
Great ideas, @ProMcTagonist and @kehugter! 
Think also that it's important to hide the zoom slider and buttons and make it visible only if you need it.

**Comment by agilob** [comment]
I wonder how will implement it...

**Comment by ProMcTagonist** [comment]
UX is not bikeshedding. Features are not bikeshedding. 

Arguing about the color of the sliders would be bikeshedding. 

Allowing users to import a picture of any size or shape and zoom/crop it easily within qTox, greatly increasing the accessibility of avatars, is not bikeshedding.

**Comment by agilob** [comment]
> Allowing users to import a picture of any size or shape and zoom/crop it easily within qTox,

so you want a picture editor inside qtox?

> greatly increasing the accessibility of avatars

would better waste time on increasing quality of audio filter, detect more video devices, fix plenty of issues with chat area. Most people on my contact list don't even have avatar, so I find this issue on the bottom of `P-low`. bikeshedding is not a kind of a feature, it's about meaning of the feature and time spent on it (discussion).

**Comment by ovalseven8** [comment]
No, it's not bikeshedding.

**Comment by tux3** [comment]
What is even the point of a "bikeshedding" tag? This isn't helpful.

**Comment by zetok** [comment]
> What is even the point of a bikeshedding tag? This isn't helping.

I agree.

**Comment by ovalseven8** [comment]
Hopefully somebody can implement it. :)

Question:
What is very important concerning this issue is that the visible area will be the same for every client. But is that really assured at the moment?
**Additionally, the client should only save the visible area then, what's not the case at the moment:**

![Bob's original avatar](https://cloud.githubusercontent.com/assets/8258609/10416693/048f39f2-7020-11e5-9032-21064576b629.GIF)
![Bob's Avatar in qTox](https://cloud.githubusercontent.com/assets/8258609/10416696/367a8d90-7020-11e5-95c2-93ee5d15fa49.png)
![Bob's avatar in /home/.config/tox/avatars](https://cloud.githubusercontent.com/assets/8258609/10559389/2354e614-74f0-11e5-8e5f-afabfd144dd2.png)

We need a clear agreement between the Tox Clients how to handle this.
@tux3 @subliun @grayhatter @Jfreegman @irungentoo @dvor

**Comment by dvor** [comment]
Well, the client should just save square photo, it will solve all the problems

**Comment by ovalseven8** [comment]
Yes, the format of the avatars in every Tox Client must be square as well.

And concerning this issue, qTox would have to save only the visible area, what's square as well. And then the visible area of the avatar would be the same, no matter which client you're using.

**Comment by ProMcTagonist** [comment]
> the format of the avatars in every Tox Client must be square as well.

1) Ugly, and
2) not needed at all.

**Comment by suhr** [comment]
3) pure bikeshedding

**Comment by GrayHatter** [comment]
No, this is stupid... you don't need permission from uTox to decide how you want to crop avatars... You decide how you want avatars to look, you crop as needed, and then you send the CROPPED avatar to your contacts. YOU DO NOT, send uncropped avatars to another client, and then tell them how you expect them to crop it. That's stupid...

**Comment by ProMcTagonist** [comment]
@GrayHatter he just wants clients to send square avatars to avoid disagreements about what to display if a tall rectangular avatar overflows, it's not really relevant to the issue of allowing fine-grained avatar control in-client before sending the image
